app/components/billing-header/component.js

```diff
@@ -3,19 +3,18 @@ import Ember from 'ember';
 const reads = Ember.computed.reads;
 
 export default Ember.Component.extend({
-  organization: null,
   billingDetail: null,
 
   isLoaded: Ember.computed.notEmpty('billingDetail'),
-  plan: reads('organization.plan'),
+  plan: reads('billingDetail.plan'),
   paymentMethodName: reads('billingDetail.paymentMethodName'),
   paymentMethodDisplay: reads('billingDetail.paymentMethodDisplay'),
   nextInvoiceDate: reads('billingDetail.nextInvoiceDate'),
 
   init() {
```

Comment:
Since you are now loading billingDetail in a model hook, this init() logic is not necessary.  I would just use the organization and billingDetails object that are already loaded in the route.